### PR TITLE
docs(type-layout): refactor to better support `addon-docs`

### PR DESCRIPTION
**Pull request by SimonFinney**
## Affected issue

Related to #741

## Proposed change

- Reorder component declarations in `TypeLayout.js`
- Set `className` `defaultProp` to `null` to prevent empty string from being generated
- Rewrite prop documentation content to better align with Carbon's

### CODE REVIEW

File: src/components/TypeLayout/TypeLayout.js [hunk 1]
```diff
@@ -56,67 +47,76 @@ const TypeLayout = ({
...
+const defaultProps = {
   children: null,
-  size: 'md',
+  className: null,
 };
 
-const { children, className } = propTypes;
-
-const borderPropType = bool;
+const size = 'md';
```
Comment (jendowns) [inline src/components/TypeLayout/TypeLayout.js:90]: Hey I was wondering why this is abstracted. I'm seeing it only used in the propTypes and default propType below, right? To me this abstraction doesn't add anything, and just takes something away (i.e., immediately understanding what the default value is & what one of the `size` prop options are). Would you consider removing this & referencing `'md'` directly?

File: src/components/TypeLayout/TypeLayout.js [hunk 2]
```diff
@@ -19,19 +19,10 @@ import {
...
-  >
-    {children}
-  </StructuredListCell>
-);
-
 // TODO: `2.x` - Remove deprecated prop `bordered`.
 
 /**
- * Type layout component.
+ * Type layouts provide a means to orderly layout terms and definitions.
```
Comment (jendowns) [inline src/components/TypeLayout/TypeLayout.js:25]: I think this is a typo here? 

Perhaps:

```suggestion
 * Type layouts provide an orderly layout of terms and definitions.
```

I think it's just the "orderly" part that seems like the typo but feel free to go with whatever you prefer! 👍

File: src/components/TypeLayout/TypeLayout.js [hunk 3]
```diff
@@ -56,67 +47,76 @@ const TypeLayout = ({
...
+
+  /** Provide an optional class to be applied to the containing node */
+  className: string,
 };
 
-TypeLayout.defaultProps = {
-  border: false,
-  bordered: null,
-  className: '',
+const defaultProps = {
```
Comment (jendowns) [inline src/components/TypeLayout/TypeLayout.js:85]: Why is this being abstracted here? I think it's only used in the default props below?

To me this makes things harder to scan and comprehend. Would you consider moving this information down where it is used?

**Review by jendowns — COMMENTED** [review]
Thank you @SimonFinney -- please let me know if you have any questions about this feedback 👍

**Comment by jendowns** [comment]
Oh also, would you consider renaming this PR to a `docs` type? My impression of this PR is that is a storybook documentation update, primarily, so technically a `refactor` is meant for a bigger release than intended?

File: src/components/TypeLayout/TypeLayout.js (hunk 3 above)
Comment (SimonFinney) [reply src/components/TypeLayout/TypeLayout.js:85]: Good point! I'll move it down

File: src/components/TypeLayout/TypeLayout.js (hunk 1 above)
Comment (SimonFinney) [reply src/components/TypeLayout/TypeLayout.js:90]: Done!

**Review by jendowns — APPROVED** [review]
Thank you @SimonFinney 👍

**Pull request merged**